serialization/time_only.go

```diff
@@ -1,14 +1,15 @@
 package serialization
 
 import (
-	"errors"
+	"fmt"
 	"strings"
 	"time"
 )
 
 // TimeOnly is represents the time part of a date time (time) value.
 type TimeOnly struct {
-	time time.Time
+	time      time.Time
+	precision int // number of decimal places for nanoseconds

```

Comment:
since the precision is only needed only at serialization time/string method, wouldn't it make more sense for it to be an argument instead?